### DISCUSSION TITLE
Get Saved Album Tracks using rangeTracks based on release date

### DISCUSSION THREAD

**Discussion by BuckeyePay** [Chimildic/goofy#182] [Помощь с алгоритмом] [answered]
I am having a disconnect filtering my saved album tracks based on album release date. I imagine it is a fairly simple script, but I am having a difficult time getting it to work. Any suggestions?

**Comment by Chimildic** [comment, marked as answer]
Show the code and write your description about fail/expected behavior

upd: added fix, try to update goofy with new code

**Comment by BuckeyePay** [comment]
```js
function BeforeKidsMix() {
  let tracks = Source.getSavedAlbumTracks()
  Filter.rangeTracks(tracks, {
    album: {
      release_date: { startDate: new Date('1950.02.29'), endDate: new Date('1988.02.29') }
    }
  })
  Playlist.saveWithReplace({
    name: 'Before Kids Mix',
    tracks: tracks,
  })
}
```
```
Execution log
12:20:44 PM	Notice	Execution started
12:20:46 PM	Error	
TypeError: Cannot read property 'id' of undefined
(anonymous)	@ library.gs:2686
findIdsOfUncachedObj	@ library.gs:2675
cache	@ library.gs:2670
getCache	@ library.gs:2662
rangeTracks	@ library.gs:855
BeforeKidsMix	@ BeforeKids.gs:6
```

**Reply by Chimildic** [reply]
Added fix, try to update goofy with new code

**Comment by BuckeyePay** [comment]
thank you.

**Comment by BuckeyePay** [comment]
Since I updated the library, I am receiving this error from the updateRecentTracks trigger

<html>
<body>


Head | updateRecentTracks_ | Time-Driven | Apr 26, 2022, 12:48:00 PM | 1.504 s | Failed
-- | -- | -- | -- | -- | --
Cloud logsApr 26, 2022, 12:48:01 PMErrorScript function not found: updateRecentTracks_REFRESH
Head | updateRecentTracks_ | Time-Driven | Apr 26, 2022, 12:33:00 PM | 1.088 s | Failed
Head | updateRecentTracks_ | Time-Driven | Apr 26, 2022, 12:18:00 PM | 2.004 s | Failed
Head | updateRecentTracks_ | Time-Driven | Apr 26, 2022, 12:03:00 PM | 1.635 s | Failed
Head | updateRecentTracks_ | Time-Driven | Apr 26, 2022, 11:48:00 AM | 1.384 s | Failed
Head | updateRecentTracks_ | Time-Driven | Apr 26, 2022, 11:33:00 AM | 0.908 s | Failed
Head | updateRecentTracks_ | Time-Driven | Apr 26, 2022, 11:18:00 AM | 1.122 s | Failed
Head | updateRecentTracks_ | Time-Driven | Apr 26, 2022, 11:03:00 AM | 1.066 s | Failed
Head | updateRecentTracks_ | Time-Driven | Apr 26, 2022, 10:48:00 AM | 1.634 s | Failed
Head | updateRecentTracks_ | Time-Driven | Apr 26, 2022, 10:33:00 AM | 1.455 s | Failed
Head | updateRecentTracks_ | Time-Driven | Apr 26, 2022, 10:18:00 AM | 2.094 s | Failed
Head | updateRecentTracks_ | Time-Driven | Apr 26, 2022, 10:03:00 AM | 1.232 s | Failed
Head | updateRecentTracks_ | Time-Driven | Apr 26, 2022, 9:48:00 AM | 2.514 s | Failed
Head | updateRecentTracks_ | Time-Driven | Apr 26, 2022, 9:33:00 AM | 0.996 s | Failed
Head | updateRecentTracks_ | Time-Driven | Apr 26, 2022, 9:18:00 AM | 0.907 s | Failed


</body>
</html>

**Reply by Chimildic** [reply]
You need to go [triggers](https://script.google.com/home/triggers), then menu > errors. Show the logs here.

**Comment by BuckeyePay** [comment]
Not sure this even matters.

**Reply by BuckeyePay** [reply]
figured it out. thanks.
